controllers/k8sgpt_controller.go

```diff
@@ -133,8 +137,25 @@ func (r *K8sGPTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 
 	// If the deployment is active, we will query it directly for analysis data
 	if deployment.Status.ReadyReplicas > 0 {
-		// Get the K8sGPT client
-		response, err := r.K8sGPTClient.ProcessAnalysis(deployment, k8sgptConfig)
+		// Check if the client exists in the map
+		if _, ok := r.k8sGPTClients[k8sgptConfig.Name]; !ok {
+			// Create a new client
+			var address string
+			if os.Getenv("LOCAL_MODE") != "" {
+				address = "localhost:8080"
+			} else {
+				address = os.Getenv("K8SGPT_API_HOST")
```

Comment:
What happens if this variable is not set?

Reply:
What should the new slug be: 
```
	address = fmt.Sprintf("http://%s.%s:8080/analyze", deployment.Name, deployment.Namespace)
```

?

Reply:
Yes, or something like this :

`address = fmt.Sprintf("http://%s.%s:8080/analyze", "k8sgpt", deployment.Namespace)`

What do you think ?

Reply:
Is that the right API for the grpc impl? I thought usually it would be `v1/analyze` etc?

Reply:
No good spot ! Let me check

Reply:
We only need something like this : `address = fmt.Sprintf("%s.%s:8080", "k8sgpt", deployment.Namespace)`

Reply:
updated